addons/hr_timesheet/hr_timesheet.py

```diff
@@ -19,79 +19,53 @@
 #
 ##############################################################################
 
-import time
+from openerp import models, api, fields, exceptions
 
-from openerp.osv import fields
-from openerp.osv import osv
 from openerp.tools.translate import _
-import openerp
-from openerp.exceptions import UserError
 
-class hr_employee(osv.osv):
-    _name = "hr.employee"
+class hr_employee(models.Model):
     _inherit = "hr.employee"
-    _columns = {
-        'product_id': fields.many2one('product.product', 'Product', help="If you want to reinvoice working time of employees, link this employee to a service to determinate the cost price of the job."),
-        'journal_id': fields.many2one('account.analytic.journal', 'Analytic Journal'),
-        'uom_id': fields.related('product_id', 'uom_id', type='many2one', relation='product.uom', string='Unit of Measure', store=True, readonly=True)
-    }
-
-    def _getAnalyticJournal(self, cr, uid, context=None):
-        md = self.pool.get('ir.model.data')
-        try:
-            dummy, res_id = md.get_object_reference(cr, uid, 'hr_timesheet', 'analytic_journal')
-            #search on id found in result to check if current user has read access right
-            check_right = self.pool.get('account.analytic.journal').search(cr, uid, [('id', '=', res_id)], context=context)
-            if check_right:
-                return res_id
-        except ValueError:
-            pass
-        return False
 
-    def _getEmployeeProduct(self, cr, uid, context=None):
-        md = self.pool.get('ir.model.data')
+    @api.model    
+    # Reads the provided data file to return the default journal id for new employees    
+    def _get_default_analytic_journal(self):
+        return self.env.ref('hr_timesheet.analytic_journal', raise_if_not_found=False) or self.env['account.analytic.journal']
```

Comment:
raise_if_not_found being False by default, no need to retype it.

I don't understand the second part: you return a new environment ?